### PR TITLE
Optionalauth

### LINKED ISSUE #159: Make login optional
Login should be optional and user should be reminded to login if user tries to see profile information without logging in.

### PR DESCRIPTION
**Pull request by ravi-kishan**
Fixes #159 

Changes: Earlier home activity prompted to login. Now there is an option to login.

### CODE REVIEW

**Review by sheetal-g9 — APPROVED** [review]
Updated UI for login and changes in drawer work well.

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java [hunk 1]
```diff
@@ -14,18 +16,24 @@
  */
 public class AuthViewModel extends ViewModel {
 
-    private static final FirebaseAuthLiveData firebaseAuthLiveData = new FirebaseAuthLiveData();
+    private AuthRepository loginRepository;
+    private FirebaseAuthLiveData firebaseAuthLiveData;
```
Comment (dr0pdb) [inline app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java:20]: Get this using Dagger also if possible?

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java (hunk 1 above)
Comment (ravi-kishan) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java:20]: The repository is getting FirebaseAuthLiveData using dagger.The viewModel is getting repository using dagger.So viewModel is automatically getting FirebaseAthLiveData through dagger.

**Review by dr0pdb — APPROVED** [review]

File: app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java (hunk 1 above)
Comment (dr0pdb) [reply app/src/main/java/com/iitp/njack/iitp_connect/core/home/AuthViewModel.java:20]: Please improve your commit messages. :)

**Pull request merged**